homebrew-proposal.rst

```diff
@@ -0,0 +1,28 @@
+The initial design by Jon and me is the following prose:
+
+Linking Travis-CI or Buildbot to the homebrew-tap repository would save:
+   #. The release engineer from logging into the Mac Mini and testing the recipe.
+   #. A reviewer of the PR to homebrew-tap’s master logging into the Mac mini and testing the recipe.
+
+It would not provide the other benefits of Continuous Integration.
+That is, CI should be able to tell you before you merge a commit into master whether that commit will break something.
+This is not possible with the current Homebrew set up even if Homebrew builds were run from Flocker pushes, because the Homebrew recipe downloads the latest released .tar.gz from GCS and installs it.
+Unless there is a release which uploads the .tar.gz file to GCS, the Homebrew recipe will not change.
+
+Testing often with CI would test that the links to the dependencies have not gone down, and that a change to Homebrew itself does not interfere with our recipe.
+
+This led to the conclusion that this issue should be covered by an admin script (perhaps put in the current ``admin/make-homebrew-recipe``).
+This script would be run during the release process, in the place of the current Homebrew steps (including the review step).
```

Comment:
What you are suggesting is having two completely different ways of testing things on a Mac, with corresponding duplication of code, configuration, etc.:
1. Buildbot (up for review another branch)
2. Manual script

Looking at the bigger picture for the release process, right now we have two completely different ways of testing:
1. Buildbot
2. Manual testing for the release

Both in the specific case and in the general case this seems non-ideal. Better if everything ran a single way. If it's Buildbot we're using for automated testing then it's Buildbot we should be using for release testing. "The tarball isn't usually available" is true, but Buildbot can do conditional builds, and in fact already does some conditional building depending on whether you're in a release branch or not (documentation, IIRC). Presumably we can do conditional builds like "oh, this has progressed far enough in the release process I can expect X to be the case". You could also modify homebrew recipe to accept alternate URL (but not alternate checksum!) for interim testing, etc..

@tomprince any thoughts?